vnext/ReactUWP/Views/ViewViewManager.cpp

```diff
@@ -377,6 +379,8 @@ void ViewViewManager::UpdateProperties(ShadowNodeBase* nodeToUpdate, const folly
         {
           pViewShadowNode->AccessibilityRole(AccessibilityRoles::None);
         }
+
+        shouldBeControl = (pViewShadowNode->AccessibilityRole() != AccessibilityRoles::None);
       }
```

Comment:
I think this should not happen if propertyValue.isNull() is true, right?

Reply:
I think shouldBeControl needs to be true if we modify accessiblity properties, because otherwise they're no-opts without the DynamicAutomationPeer.

Reply:
But if they all go back to a state of not needing the peer why would we keep it? I don't know that the dynamic case like this would be common, but it seems odd to keep it as a Control when that's not needed. We've tried to update these conditions on properties so the result is accurate when a property is no longer bound.

---
In reply to: [295914113](https://github.com/microsoft/react-native-windows/pull/2638#discussion_r295914113) [](ancestors = 295914113)

Reply:
But isn't this method run only to update properties that have changed? If the first batch sets a property that requires a control, a later batch trying to change a different property can't decide to get rid of that control. We need a way of maintaining when we need the automationpeer.